docs/source/examples.rst

```diff
@@ -4,6 +4,150 @@
+
+Cart Pole example
+=================
+
+The Cart Pole environment is a classic control task in which the goal is to balance a pole on a cart. The environment is described by a 4-dimensional state space, which consists of the cart's position, the cart's velocity, the pole's angle, and the pole's angular velocity. The agent can take one of two actions: push the cart to the left or push the cart to the right. The episode ends when the pole falls below a certain angle or the cart moves outside of the environment's boundaries. The goal is to keep the pole balanced for as long as possible.
+
+The following example demonstrates how to train a reinforcement learning agent using Reinforced-lib and OpenAI Gym. The agent uses the Deep Q-Network (DQN) algorithm to learn how to balance the pole. The DQN algorithm is implemented in Reinforced-lib and the Cart Pole environment is provided by Gym.
+
+We start with the neccessary imports:
+
+.. code-block:: python
+
+    import gymnasium as gym
+    import optax
+    from chex import Array
+    from flax import linen as nn
+
+    from reinforced_lib import RLib
+    from reinforced_lib.agents.deep import DQN
+    from reinforced_lib.exts import Gymnasium
+    from reinforced_lib.logs import StdoutLogger, TensorboardLogger
+
+We than define the QNetwork approximator as a simple dense, feed forward neural network with a ReLU activation function:
+
+.. code-block:: python
+
+    class QNetwork(nn.Module):
+        @nn.compact
+        def __call__(self, x: Array) -> Array:
+            x = nn.Dense(256)(x)
+            x = nn.relu(x)
+            return nn.Dense(2)(x)
+
+Next, we step into the run() function, which is responsible for training the agent. We start by instantiating the Reinforced-lib, specifying the ageent as a DQN, the extension as Gymnasium, and the loggers as StdoutLogger and TensorboardLogger to log both to the console and to a Tensorboard file.
+
+.. code-block:: python
+
+    def run(num_epochs: int) -> None:
+        rl = RLib(
+            agent_type=DQN,
+            agent_params={
+                'q_network': QNetwork(),
+                'optimizer': optax.rmsprop(3e-4, decay=0.95, eps=1e-2),
+                'discount': 0.95,
+                'epsilon_decay': 0.9975
+            },
+            ext_type=Gymnasium,
+            ext_params={'env_id': 'CartPole-v1'},
+            logger_types=[StdoutLogger, TensorboardLogger]
+        )
+
+We than start the training loop, where we iterate over the number of epochs and for each epoch we run the agent in the environment. We start by resetting the environment and sampling the agent's initial action. Than we run the agent int he environment by updating the environment state with the action and sampling the next action. We continue this loop until the environment reaches a terminal state. We log the length of the epoch and continue to the next epoch.
```

Comment:
Than we run the agent int **the** environment by **performing an action in** ~~updating~~ the environment ~~state with the action~~

Reply:
than != then

"We *then* start the training loop"

"*Then,* we run the agent in the environment"
or
"*Next,* we run the agent in the environment"

Reply:
Yes, of course - I missed this typo and did not correct it in the comment.

Reply:
Thanks, I've corrected these spelling mistakes. Next time I will double check with language tools.